Sources/PackagePlugin/Plugin.swift

```diff
@@ -38,15 +40,34 @@
 extension Plugin {
     
     public static func main(_ arguments: [String]) throws {
+        
+        // Use the initial `stdout` for returning JSON, and redirect `stdout`
+        // to `stderr` for capturing freeform text.
+        let jsonOut = fdopen(dup(fileno(stdout)), "w")
+        dup2(fileno(stderr), fileno(stdout))
+        
+        // Close `stdin` to avoid blocking if the plugin tries to read input.
+        close(fileno(stdin))
```

Comment:
@compnerd or @pcbeard could use your expertise here.  I believe this all works the same on Windows if I import `ucrt`.  Should I use the underscored versions here?

Of course we should use Foundation whenever possible, but I don't see a way to do this using Foundation.